plugins/scaffolder/src/components/MultistepJsonForm/schema.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JsonObject } from '@backstage/config';
+import { FormProps } from '@rjsf/core';
+
+function isObject(value: unknown): value is JsonObject {
+  return typeof value === 'object' && value !== null && !Array.isArray(value);
+}
+
+function extractUiSchema(schema: JsonObject, uiSchema: JsonObject) {
+  const { properties } = schema;
+  if (!isObject(properties)) {
+    return;
+  }
+  for (const propName in properties) {
+    if (!properties.hasOwnProperty(propName)) {
+      continue;
+    }
+    const schemaNode = properties[propName];
+    if (!isObject(schemaNode)) {
+      continue;
+    }
+
+    if (schemaNode.type === 'object') {
+      const innerUiSchema = {};
+      uiSchema[propName] = innerUiSchema;
+      extractUiSchema(schemaNode, innerUiSchema);
+    } else {
+      for (const innerKey in schemaNode) {
+        if (!schemaNode.hasOwnProperty(innerKey)) {
+          continue;
+        }
+        const innerValue = schemaNode[innerKey];
```

Comment:
would be cleaner with `for (const [innerKey, innerValue] of Object.entries(schemaNode))`

Reply:
the delete doesn't pose a problem for that does it?

Reply:
Yeah we tried, TypeScript said bleh